src/cheri/zyseal.adoc

```diff
@@ -0,0 +1,104 @@
+introduction and elimination of sealed capability forms,
+in keeping with CHERI's _principle of intentional use_.
+
+This extension first introduces a fundamentally new _kind_ of capabilities,
+"**type** capabilities",
+whose address space and borne authority ranges not over _memory_
+but rather <<sec_cap_type>>-s.
+As subsequently detailed, these capabilities, and their new permissions,
+will serve as authorizing capabilities to new instructions
+which perform transformations of other capabilities' <<sec_cap_type>> fields:
+
+* Constructing a sealed capability with type `T` from an unsealed capability
+  requires the authority to seal at type `T`, and
+
+* Constructing an unsealed capability from a sealed capability with type `T`
+  requires the authority to unseal at type `T`.
+
+This extension does not define "type conversion" transformations
+directly between sealed capability types.
+
+=== Usable <<sec_cap_type>> Values Are Encoding Specified
+
+The capabilities used to mediate (un)sealing are, like memory capabilities,
+associated with an XLEN-bit address space.
+However, capability encodings will have fewer than XLEN bits devoted to storing
+<<sec_cap_type>> values.
+As such, encodings will specify what <<sec_cap_type>> values can be used to seal
+capabilities
+(recall that encodings _must_ support representing unsealed capabilities).
+The remainder of the address space described by type capabilities
+is available for software use.
+
+=== Single Address Space Encodings
+
+Capability encodings are permitted to conflate memory and type address spaces,
+such that one capability may authorize both memory access to a location and
+(un)sealing with a type of equal numeric value.
+Indeed, the encoding of <<app_cap_description>> is one such encoding.
+Ideally, such encodings should permit separate manipulation of (un)sealing
+permission and memory access permissions,
+so that software can segregate the address spaces even when the encoding does
+not intrinsically.
+
+=== Added Architectural Permissions (AP) Bits
+
+[#zyseal_ap_field,width="100%",options=header,halign=center,cols="2,2,5"]
+|==============================================================================
+| Permission         | Type                        | Comment
+| <<zyseal_se_perm>> | <<sec_cap_type>> permission | Grants sealing authority
+| <<zyseal_us_perm>> | <<sec_cap_type>> permission | Grants unsealing authority
+|==============================================================================
+
+[#zyseal_se_perm,reftext="SE-permission"]
+Seal Permission (SE):: Permit the bearer to <<YSEAL>> capabilities
+at the in-bound types of this capability.
+
+[#zyseal_us_perm,reftext="US-permission"]
+Unseal Permission (US):: Permit the bearer to <<YUNSEAL>> capabilities
+at the in-bound types of this capability.
+
+=== Interaction with <<CLRPERM>> and <<GCPERM>>
+
+.Extended capability permissions bit field (see <<acperm_bit_field>>)
```

Comment:
Is it a good idea to show an example encoding for SE and US perms?

Reply:
That sounds like it would be quite a lot of pain - maybe easier to just include the LaTex (Tikz) image?

Reply:
> That sounds like it would be quite a lot of pain - maybe easier to just include the LaTex (Tikz) image?

Too late.